### DISCUSSION TITLE
Is every page being cached ?

### DISCUSSION THREAD

**Discussion by zwelhtetyan** [vercel/next.js#52953] [App Router]
```tsx
        <Image
          src={coverImages[getRandomNumber(0, 2)].src}
          alt="cover photo"
          width={750}
          height={400}
          priority
          className="h-48 w-full object-cover object-center xs:h-60"
        />

```

I like to pick random photo on every time the page is reload. 
The above code is working fine in development, but not in production.

**Comment by icyJoseph** [comment]
You can see when the build is done, there's a printed summary. It has some symbols:

```console
λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
```

For example:

```console
Route (app)                                     Size     First Load JS
┌ ○ /                                           20.9 kB         112 kB
├ ○ /_not-found                                 0 B                0 B
├ ○ /blog                                       805 B          97.9 kB
├ ● /blog/[slug]                                19.3 kB         116 kB
├   ├ /blog/state-with-dependency-tracking
├   ├ /blog/use-reactive-local-storage
├   ├ /blog/aoc-2021-day-1
├   └ /blog/who-am-i
└ λ /views                                      0 B                0 B
```

This has deep meaning because in RSC, the default behavior is to pre-render as much as possible. Even Route Handlers can be statically compiled. 

Read point 4 here: https://demystifying-rsc.vercel.app/notes/

Anyhow, you might want to opt into `export const dynamic = 'force-dynamic'` but that won't prevent the client side server component cache from saving that route. https://www.inngest.com/blog/5-lessons-learned-from-taking-next-js-app-router-to-production#1-understand-the-two-caches

**Comment by zwelhtetyan** [comment]
How could my expected behavior(`change random image on reload`) be achieved?

**Reply by masterbater** [reply]
@icyJoseph can you please answer since I have some similar issue where I want to add today's date as props like

```jsx
<Component today={new Date()}/>
```
This always render the date when it was built. Using export true
